packages/sdk/src/utils/assets.ts

```diff
@@ -0,0 +1,141 @@
+import type { Address } from 'viem'
+import { parseUnits } from 'viem'
+
+import type { SupportedChainId } from '@/constants/supportedChains.js'
+import { getTokenAddress, SUPPORTED_TOKENS } from '@/supported/tokens.js'
+
+/**
+ * Asset identifier - can be a symbol (like 'usdc') or address
+ */
+export type AssetIdentifier = string | Address
+
+/**
+ * Resolved asset information
+ */
+export interface ResolvedAsset {
+  address: Address
+  symbol: string
+  decimals: number
+}
+
+/**
+ * Resolve asset identifier to address and metadata
+ * @param asset - Asset symbol (e.g. 'usdc') or address
+ * @param chainId - Chain ID to resolve for
+ * @returns Resolved asset information
+ * @throws Error if asset is not supported or found
+ */
+export function resolveAsset(
```

Comment:
I do think the `TokenRegistry` + `Token` class abstraction helps simplify and make this code a bit more maintainable, but I also get if we dont want the extra overhead of classes. Definitely not a requirement to refactor now but we should keep an eye on this and see how it progresses. If this file starts to get pretty big or overly complex we should consider bringing those abstractions back.